src/components/results/ResultsList.vue

```diff
@@ -1,130 +1,198 @@
+import { useResultsStore } from "../../stores/results";
+import { ref, watch } from "vue";
+import { Candidate } from "../../../dto/candidates";
+import { Result } from "../../../dto/results";
 
 const router = useRouter();
+const { getAllCandidates } = useCandidateStore();
+const { getResultsForCandidate } = useResultsStore();
+
+const candidatesList = ref<Candidate[]>([]);
+const selectedCandidate = ref("");
+const isCandidatesVisible = ref(false);
+const choosedCandidates = ref<Candidate[]>([]);
+const quizResults = ref<Result[]>([]);
+const isLoaderVisible = ref(true);
 
-function pushRoute(id: string) {
+async function getAllCandidatesData() {
+  try {
+    isLoaderVisible.value = true;
+    const {
+      data: { candidates },
+    } = await getAllCandidates();
+    candidatesList.value = candidates;
+  } catch (e) {
+    console.log(e);
+  }
+  isLoaderVisible.value = false;
+}
+getAllCandidatesData();
+
+async function getResultsForCandidateData(candidateId: number) {
+  try {
+    isLoaderVisible.value = true;
+    const { data } = await getResultsForCandidate(candidateId);
+    quizResults.value = data;
+  } catch (e) {
+    console.log(e);
+  }
+  isLoaderVisible.value = false;
+}
+
+watch(selectedCandidate, newCandidate => {
+  choosedCandidates.value = candidatesList.value.filter(candidate =>
+    candidate.username.includes(newCandidate),
+  );
+});
+function setCandidate(user: Candidate) {
+  quizResults.value = [];
+  isCandidatesVisible.value = false;
+  selectedCandidate.value = user.username;
+  getResultsForCandidateData(user.id);
+}
+
+function pushRoute(candidateId: string, resultId: string) {
   router.push({
     name: "singleResult",
-    params: { id },
+    params: {
+      candidateId,
+      resultId,
+    },
   });
 }
 </script>
 
 <template>
   <div class="container mt-3">
-    <h2 class="text-primary text-center">Quiz Results</h2>
-    <ul class="list-unstyled">
-      <li
-        v-for="item in quizResults"
-        :key="item.id"
-        class="border border-light mt-4 p-2 rounded-3 text-secondary mx-auto shadow-sm ps-4"
-        @click="pushRoute(item.id)"
+    <h2 class="text-primary text-center text-md-start mb-md-2">Quiz Results</h2>
+    <div
+      class="alert alert-warning"
+      role="alert"
+    >
+      <font-awesome-icon
+        icon="fa-solid fa-circle-info"
+        class="fs-6"
+      />
+      Quiz results are grouped by Candidates. For displaying results, please,
+      choose candidate from list
+    </div>
+    <div class="col-12 w-100 position-relative">
+      <h2 class="text-primary text-center text-md-start mt-2">
+        Choose candidate
+      </h2>
+      <input
+        id="candadidateInput"
+        v-model="selectedCandidate"
+        class="form-control"
+        placeholder="Enter username to search..."
+        @focusin="isCandidatesVisible = true"
+      />
+      <div
+        v-if="selectedCandidate.length >= 1 && isCandidatesVisible"
```

Comment:
I still didn't get this check for one symbol. What is it for? Isn't just `v-if="selectedCandidate && isCandidatesVisible"` enough?